### PR TITLE
Adds Movable Storage Windows

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Title. Will expand this description when it's not 1:07 AM.

### CODE REVIEW

**Comment by MarinaGryphon** [comment]
!review

File: code/_onclick/hud/movable_screen_objects.dm [hunk 1]
```diff
@@ -89,6 +92,34 @@
...
+	if(!movable)
+		return
+	if(istype(master, /obj/item/weapon/storage/))
+		var/obj/item/weapon/storage/S = master
+		S.space_orient_objs()
+
+/obj/screen/movable/storage/Click()
+	if(!usr.canClick())
+		return 1
+	if(usr.stat || usr.paralysis || usr.stunned || usr.weakened)
```
Comment (skull132) [inline code/_onclick/hud/movable_screen_objects.dm]: `use_check()` with specific flags might be a better idea.

File: code/game/objects/items/weapons/storage/storage.dm [hunk 2]
```diff
@@ -100,27 +99,16 @@
...
-	user.client.screen -= storage_end
-	user.client.screen -= closer
-	user.client.screen -= contents
-	user.client.screen += closer
-	user.client.screen += contents
+	for(var/obj/item/I in src)
+		if(I.on_found(user))
+			return
+	
+	user.client.screen |= closer
```
Comment (skull132) [inline code/game/objects/items/weapons/storage/storage.dm]: Any particular reason for the `|=` operators? It could be wasted work.

Comment (MarinaGryphon) [reply code/game/objects/items/weapons/storage/storage.dm]: It originally used `-=` then `+=`, which is, as far as I know, syntactically equivalent.

Comment (skull132) [reply code/game/objects/items/weapons/storage/storage.dm]: uh. Absolutely not lmao.
`|=` is: "If not found in list, add." It does not remove. It _does_ however scan the entire list, looking to see if it's already in it first.

**Comment by alsoandanswer** [comment]
https://i.imgur.com/T9eJjOE.gifv

Gif for people to see whatever the heck this PR is about.

File: code/game/objects/items/weapons/storage/storage.dm (hunk 2 above)
Comment (MarinaGryphon) [reply code/game/objects/items/weapons/storage/storage.dm]: -= then += is the exact same as |= in that it adds it while preventing duplicates. It is functionally the exact same.

File: code/game/objects/items/weapons/storage/storage.dm [hunk 3]
```diff
@@ -161,37 +142,33 @@
...
-	return
-
 //This proc draws out the inventory and places the items on it. It uses the standard position.
 /obj/item/weapon/storage/proc/slot_orient_objs(var/rows, var/cols, var/list/obj/item/display_contents)
 	var/cx = 4
 	var/cy = 2+rows
-	src.boxes.screen_loc = "4:16,2:16 to [4+cols]:16,[2+rows]:16"
+
+	var/list/screen_loc_xyfull = text2list(storage.screen_loc," to ")
+	var/list/screen_loc_xy = list(screen_loc_xyfull[1])
```
Comment (MarinaGryphon) [inline code/game/objects/items/weapons/storage/storage.dm]:
```suggestion
	var/list/screen_loc_xy = text2list(screen_loc_xyfull[1], ",")
```

File: code/_onclick/hud/movable_screen_objects.dm (hunk 1 above)
Comment (MarinaGryphon) [reply code/_onclick/hud/movable_screen_objects.dm]:
```suggestion
	if(use_check(usr, USE_ALLOW_NON_ADV_TOOL_USR))
```
Monkeys et al. should still be able to remove things from bags.

**Comment by NonQueueingMatt** [comment]
How does this behave with misclicks? This is gonna be really annoying if it triggers on every missed click or clickdrag.

**Comment by NonQueueingMatt** [comment]
Merge conflicts, unaddressed review, reopen when ready.

**Pull request closed without merging**